### DISCUSSION TITLE
What is the precise difference between Whitelist and "not Blacklist"?

### DISCUSSION THREAD

**Discussion by Michael-Breu-UIbk** [ls1intum/Ares#115] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
I have problems to understand, when I have to explicitely whitelist a certain feature.

It seems, if something is not whitelisted and not blacklisted it raises an error.

**Describe the solution you'd like**
Document in the readme.md what is the precise consequence, not to whitelist something.

**Describe alternatives you've considered**
None

**Additional context**

**AJTS-Version you are using**
current.

**Comment by MaisiKoleni** [comment, marked as answer]
I know, this got messier over time the more options were needed. It might be necessary to redesign a lot of that, as it is not always logical.

I will add more information to the README when I have the time.

Usually AJTS will block paths unless they are whitelisted, and blacklist beats whitelist.

For packages, it is the other way round, they are whitelisted per default. But this is an experimental feature that does not work properly yet.

WhitelistClass makes AJTS trust a class that is otherwise not trusted, an opposite annotation does not exist.

AddTrustedPackage is the package equivalent of WhitelistClass.

AllowThreads and AllowLocalPort are completely on their own.
